Icons 11

merinsky opened the pull request:
### Describe the pull request:
- [x ] Code cleanup

### Pull request long description:
Icons code cleanup.
First step to enable work on icons and usage of your current GUI icons in Subsurface, where meaningfull.
Up to this point, nothing is changed in GUI.

### Changes made:
1) Use relative icon path.
2) Using icon aliases from Qt Resource System.

neolit123 commented:
> This changes icon aliases to one, easy grep-able style.

sorry, NAK. i don't think you've understood the points from the previous pull request.
some of the changes are good, some are bad.

merinsky commented:
Pleas ask one question. Is better current state or what I'm proposing?

I understand pretty well, just don't agree with you. Great number of icons are not specific for widget. For example `map-widget/qml/icons/mapwidget-zoom-in.png`. Is there any reason map-widget should have it's own zooming icon? No. Zooming is not exclusive for map-widget. User zooms in Firefox, Gimp, Inkscape... There is one common action - zooming. It shoud share the same icon through all applications. Because than, it is eaysy recognizable "what is this icon (button) doing?". The same for opening menus (hamburger icon), adding or removing items from a list. And so on.

In some very specific cases, widgets may need their own icons, but I believe it is very seldom and problematic case.

I didn't invent it. I'm only using, what smarter people figured out. And it seems to work pretty well.

If you have any specific feedback, please write it here @neolit123 . NAK without reason is useless. I spent a lot of time working on this.

neolit123 commented:
it is clear that you've spent time on this, but the comments from the previous PR were ignored.

> I understand pretty well, just don't agree with you. Great number of icons are not specific for widget. For example map-widget/qml/icons/mapwidget-zoom-in.png. Is there any reason map-widget should have it's own zooming icon?

until an icon is re-used in multiple locations in the code base it should remain with a localized context in the resources.

> If you have any specific feedback, please write it here @neolit123 . NAK without reason is useless. I spent a lot of time working on this.

reasons were explained; the resource aliases need context added to them not removed or **paraphrased**.

also, resource file renaming is in order if you change certain aliases.

merinsky commented:
> comments from the previous PR were ignored

They were not. I renamed aliases for map types, partial pressures, map markers as you wished. I tried and did found some better names for several others. I avoided any kind of sorting and mixing changes in UI, cleanup, removing and adding things. And i reworked all following commits.

I'm not paraphrasing anything. Please stick to relevant feedback "_foo_ could have better name _bar_". Context is defined very clearly in **standard**, widely used way here: https://github.com/merinsky/subsurface/blob/icons_wip/icons/diving/index.theme I hope we wil not suffer _not invented here_ syndrome.

> until an icon is re-used in multiple locations in the code base it should remain with a localized context in the resources.

Icons are reused through several applications. As far as I can see, this is your only objection. Your idea is not bad, but I was investigating, how is it done. And I found that approach I used.

I will not add widget name to icon file names nor aliases generally. It is wrong idea. If you have specific case, where you think it should be done, let me know.

dirkhh commented on desktop-widgets/filterwidget.ui:
```diff
@@ -88,7 +88,7 @@
        <property name="icon">
         <iconset resource="../subsurface.qrc">
-         <normaloff>:filter-hide</normaloff>:filter-hide</iconset>
+         <normaloff>:show-icon</normaloff>:show-icon</iconset>
```
This one surprised me. I haven't looked at the icons - just the switch from 'hide' to 'show'...

merinsky replied:
Current name is misleading, because the same icon is used for show and hide actions. So this doesn't matter too much. Hopefully, we will use different icons someday. I will change this to *hide*.

dirkhh commented on profile-widget/diveeventitem.cpp:
```diff
@@ -81,19 +81,19 @@ void DiveEventItem::setupPixmap()
 		struct gasmix *mix = get_gasmix_from_event(&displayed_dive, internalEvent);
 		if (mix->he.permille)
-			setPixmap(EVENT_PIXMAP_BIGGER(":gaschangeTrimix"));
+			setPixmap(EVENT_PIXMAP_BIGGER(":gaschange-tmx-icon"));
```
tmx for trimix? I'm not familiar with that abbreviation. Let's use trimix unless there's a good reason for tmx

merinsky replied:
Sure, no problem. I used it, because current icon file-name is *Tmx_change.png* and it looks similar to *air*, *ean*, *oxy* used for the rest gaschage events.

dirkhh replied:
I didn't even notice that. I think as we clean it up, the switch to trimix might be easier to understand.

dirkhh commented on profile-widget/diveeventitem.cpp:
```diff
@@ -81,19 +81,19 @@ void DiveEventItem::setupPixmap()
+		setPixmap(EVENT_PIXMAP(":status-warning-icon"));
 	} else if (internalEvent->type == SAMPLE_EVENT_BOOKMARK) {
-		setPixmap(EVENT_PIXMAP(":flag"));
+		setPixmap(EVENT_PIXMAP(":bookmark-icon"));
```
this isn't a regular bookmark. This is really a dive flag. Not sure about this one.

merinsky replied:
You are right. I will change this to more appropriate *dive-bookmark-icon*.

dirkhh commented:
> also, resource file renaming is in order if you change certain aliases.

@neolit123 - can you explain that comment in more detail? I don't understand what you mean.

neolit123 commented:
>> also, resource file renaming is in order if you change certain aliases.

> @neolit123 - can you explain that comment in more detail? I don't understand what you mean.

having an alias point to a rather unrelated file name:

```
<file alias="bookmark-icon">icons/flag.png</file>
...
<file alias="cylinders-icon">icons/gaschange.png</file>
```

dirkhh commented:
Good point, @neolit123 - we should call this gaschange-icon because that's what it really is.

merinsky commented:
Renaming icons (changing icon file names) is on my list. I took baby steps. Proposed flle names are related to their aliases, see (still WIP): https://github.com/merinsky/subsurface/blob/icons_wip/subsurface.qrc

merinsky commented:
OK, I will rename *cylinders-icon* to *gaschange-icon*, as you are proposing.

dirkhh commented:
So I'm waiting for the promised change to "gaschange-icon" :-)

merinsky commented:
Glad to hear it :-) What is the right procedure here?
* Fixup commit?
* Force-push to this branch?
* New branch with new PR?

dirkhh commented:
Force push

On November 30, 2017 9:48:02 PM PST, merinsky <notifications@github.com> wrote:
>Glad to hear it :-) What is the right procedure here?
>* Fixup commit?
>* Force-push to this branch?
>* New branch with new PR?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/Subsurface-divelog/subsurface/pull/893#issuecomment-348407455

-- 
from my phone.

merinsky commented:
Done.

dirkhh commented:
Thanks for your patience with us :-)
